src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-share-form/smb-share-form.component.ts

```diff
@@ -0,0 +1,175 @@
+import { CephfsSubvolumeGroup } from '~/app/shared/models/cephfs-subvolume-group.model';
+
+@Component({
+  selector: 'cd-smb-share-form',
+  templateUrl: './smb-share-form.component.html',
+  styleUrls: ['./smb-share-form.component.scss']
+})
+export class SmbShareFormComponent extends CdForm implements OnInit {
+  smbShareForm: CdFormGroup;
+  action: string;
+  resource: string;
+  allFsNames: Filesystem[] = [];
+  allsubvolgrps: CephfsSubvolumeGroup[] = [];
+  allsubvols: CephfsSubvolume[] = [];
+  clusterId: string;
+
+  constructor(
+    private formBuilder: CdFormBuilder,
+    public smbService: SmbService,
+    public actionLabels: ActionLabelsI18n,
+    private nfsService: NfsService,
+    private subvolgrpService: CephfsSubvolumeGroupService,
+    private subvolService: CephfsSubvolumeService,
+    private taskWrapperService: TaskWrapperService,
+    private router: Router,
+    private route: ActivatedRoute
+  ) {
+    super();
+    this.resource = $localize`Share`;
+  }
+  ngOnInit() {
+    this.action = this.actionLabels.CREATE;
+    this.route.params.subscribe((params: { clusterId: string }) => {
+      this.clusterId = params.clusterId;
+    });
+    this.nfsService.filesystems().subscribe((data: Filesystem[]) => {
+      this.allFsNames = data;
+    });
+    this.createForm();
+  }
+
+  createForm() {
+    this.smbShareForm = this.formBuilder.group({
+      share_id: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      volume: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      subvolume_group: new FormControl(''),
+      subvolume: new FormControl(''),
+      prefixedPath: new FormControl({ value: '', disabled: true }),
+      inputPath: new FormControl('', {
+        validators: [Validators.required]
+      }),
+      browseable: new FormControl(true),
+      readonly: new FormControl(false),
+      intent: new FormControl(Intent.Present)
+    });
+  }
+
+  volumeChangeHandler() {
+    const fsName = this.smbShareForm.getValue('volume');
+    this.smbShareForm.patchValue({
+      subvolume_group: '',
+      subvolume: '',
+      prefixedPath: ''
+    });
+    if (fsName) {
+      this.getSubVolGrp(fsName);
+    }
+  }
+  getSubVolGrp(volume: string) {
+    this.smbShareForm.patchValue({
+      subvolume_group: '',
+      subvolume: ''
+    });
+    if (volume) {
+      this.subvolgrpService.get(volume).subscribe((data: CephfsSubvolumeGroup[]) => {
+        this.allsubvolgrps = data;
+      });
+    }
+  }
+
+  getSubVol() {
+    const volume = this.smbShareForm.getValue('volume');
+    const subvolgrp = this.smbShareForm.getValue('subvolume_group');
+    this.smbShareForm.patchValue({
+      subvolume: ''
+    });
+
+    if (volume && subvolgrp) {
+      this.subvolService.get(volume, subvolgrp).subscribe((data: CephfsSubvolume[]) => {
+        this.allsubvols = data;
+      });
+    }
+    this.updatePath('/volumes/' + subvolgrp);
```

Comment:
```suggestion
    this.updatePath(VOLUME_PREFIX_PATH' + subvolgrp);
```
lets add constant for `'/volumes/` and reuse everywhere else

Reply:
This field is no longer needed.